CapstoneProject/Assets/Scripts/Core/Hazards/HazardAreaDeath.cs

```diff
@@ -0,0 +1,18 @@
+using System.Collections;
+using System.Collections.Generic;
+using Core.Entity;
+using UnityEngine;
+
+public class HazardAreaDeath : MonoBehaviour
+{
+    private PlayerHealthComponent _playerHealthComponent;
+
+    private void OnTriggerEnter2D(Collider2D other)
+    {
+        if (other.GetComponent<PlayerHealthComponent>())
+        {
+            _playerHealthComponent = other.GetComponent<PlayerHealthComponent>();
```

Comment:
just a note, doesn't need to be updated but for future, you do the same get component here, twice. I'd just assigned the value outside the if first and just check if the _playerhealth is null or not before calling the kill function. Saves one call to getcomponent.